File: K47 User Manual/2_light_and_infrared_sensors/15_obstacleAvoidance/Code/Python/obstacle.py
#!/usr/bin/env python
import RPi.GPIO as GPIO

ObstaclePin = 11
LedPin = 16

def setup():
	GPIO.setmode(GPIO.BOARD)       # Numbers GPIOs by physical location
	GPIO.setup(ObstaclePin, GPIO.IN, pull_up_down=GPIO.PUD_UP)
        GPIO.setup(LedPin, GPIO.OUT)
		
def loop():
	while True:
		if (0 == GPIO.input(ObstaclePin)):
		    print "Barrier is detected !"
                    GPIO.output(LedPin, True)
                else:
                    GPIO.output(LedPin, False)
			

def destroy():
	GPIO.cleanup()                      # Release resource

if __name__ == '__main__':     			# Program start from here
	setup()
	try:
		loop()
	except KeyboardInterrupt:  
		destroy()

File: K47 User Manual/2_light_and_infrared_sensors/15_obstacleAvoidance/Code/C/obstacle.c
#include <wiringPi.h>
#include <stdio.h>

#define ObstaclePin 0
#define LedPin		4

int main(void)
{
	if(wiringPiSetup() == -1)
	{ 
		printf("setup wiringPi failed !\n");
		return -1; 
	}
	
	pinMode(LedPin, OUTPUT);
	while(1)
	{
		if(0 == digitalRead(ObstaclePin))
		{
			printf("Barrier detected!\n");
			digitalWrite(LedPin, HIGH);
		}
		else
		{
			digitalWrite(LedPin, LOW);
		}
	}
	return 0;
}
